GitUI/UserControls/InteractiveGitActionControl.cs

```diff
@@ -0,0 +1,265 @@
+
+            string actionStr = "";
+
+            switch (_action)
+            {
+                case GitAction.Bisect:
+                    actionStr = _bisect.Text;
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+                case GitAction.Rebase:
+                    actionStr = _rebase.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+                case GitAction.Merge:
+                    actionStr = _merge.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    break;
+                case GitAction.Patch:
+                    actionStr = _patch.Text;
+                    ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                    ButtonContainer.Controls.Add(AbortButton);
+                    ButtonContainer.Controls.Add(MoreButton);
+                    break;
+                case GitAction.None:
+                    // can only get here if hasConflicts so add resolve button
+                    ButtonContainer.Controls.Add(ResolveButton);
+                    break;
+            }
+
+            TextLabel.Text = (_action == GitAction.None) ?
+                _conflictsMessage.Text :
+                string.Format(
+                    _hasConflicts ?
+                        _progressWithConflictsMessage.Text :
+                        _progressMessage.Text,
+                    actionStr);
+
+            Visible = true;
+        }
+
+        private void ResolveButton_Click(object sender, EventArgs e)
+        {
+            Form?.UICommands.StartResolveConflictsDialog(this);
+        }
+
+        private void ContinueButton_Click(object sender, EventArgs e)
+        {
+            if (!(Form is GitModuleForm))
+            {
+                return;
+            }
+
+            switch (_action)
+            {
+                case GitAction.Rebase:
+                    FormProcess.ShowDialog(Form, GitCommandHelpers.ContinueRebaseCmd());
+                    break;
+                case GitAction.Merge:
+                    FormProcess.ShowDialog(Form, GitCommandHelpers.ContinueMergeCmd());
+                    break;
+                case GitAction.Patch:
+                    FormProcess.ShowDialog(Form, GitCommandHelpers.ResolvedCmd());
+                    break;
+                default:
+                    return;
+            }
+
+            Form.UICommands.RepoChangedNotifier.Notify();
+        }
+
+        private void AbortButton_Click(object sender, EventArgs e)
+        {
+            if (!(Form is GitModuleForm))
```

Comment:
```suggestion
            if (Form is null)
```
And other places too